monitor/services/src/main/java/org/datacleaner/monitor/server/controllers/ReferenceDataController.java

```diff
@@ -19,59 +19,361 @@
+        return createRDMStream(tenant, ReferenceDataType.STRING_PATTERN, true).collect(Collectors.toList());
+    }
+
+    @RolesAllowed(SecurityRoles.VIEWER)
+    @RequestMapping(value = "/stringPattern/{name}", method = { RequestMethod.GET, RequestMethod.HEAD },
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public StringPatternModel getStringPattern(@PathVariable("tenant") final String tenant,
+            @PathVariable("name") final String name, final HttpServletRequest request) {
+        final StringPattern stringPattern = getReferenceDataCatalog(tenant).getStringPattern(name);
+
+        if (stringPattern == null) {
+            logger.warn("Could not find string pattern \"" + name + "\"");
+            throw new NoSuchResourceException();
+        }
+
+        if (request.getMethod().equals(RequestMethod.HEAD.name())) {
+            return null;
+        }
+
+        if (stringPattern instanceof SimpleStringPattern) {
+            return new StringPatternModel(name, ((SimpleStringPattern) stringPattern).getExpression(),
+                    StringPatternModel.PatternType.STRING);
+        } else if (stringPattern instanceof RegexStringPattern) {
+            final RegexStringPattern regexStringPattern = (RegexStringPattern) stringPattern;
+            return new StringPatternModel(name, regexStringPattern.getExpression(),
+                    regexStringPattern.isMatchEntireString() ? StringPatternModel.PatternType.REGEX_MATCH_ENTIRE_STRING
+                            : StringPatternModel.PatternType.REGEX);
+        } else { // RegexSwap. Treat as regex.
+            final RegexSwapStringPattern regexStringPattern = (RegexSwapStringPattern) stringPattern;
+            return new StringPatternModel(name, regexStringPattern.getRegex().getExpression(),
+                    StringPatternModel.PatternType.REGEX);
+        }
+    }
+
+    @RolesAllowed(SecurityRoles.CONFIGURATION_EDITOR)
+    @RequestMapping(value = "/stringPattern/{name}", method = { RequestMethod.PUT },
+            consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity putStringPattern(@PathVariable("tenant") final String tenant,
+            @PathVariable("name") final String name, @RequestBody final StringPatternModel stringPatternModel) {
+
+        final StringPattern stringPattern;
+        switch (stringPatternModel.getPatternType()) {
+        case STRING:
+            stringPattern = new SimpleStringPattern(name, stringPatternModel.getPattern());
+            break;
+        case REGEX:
+            stringPattern = new RegexStringPattern(name, stringPatternModel.getPattern(), false);
+            break;
+        case REGEX_MATCH_ENTIRE_STRING:
+            stringPattern = new RegexStringPattern(name, stringPatternModel.getPattern(), true);
+            break;
+        default:
+            throw new IllegalArgumentException("No such string pattern type");
```

Comment:
Not really a problem, it's an enum so it should never happen, mostly here because Java doesn't realize that all branches are already covered, making it complain about possibly uninitialized stringPattern below. OTOH, it's not really a tough task :)